src/models/supplemental_constructors.jl

```diff
@@ -46,7 +46,40 @@ function Deterministic(component::Component,
         initial_time : Dates.Hour(1) : initial_time + resolution * (time_steps-1),
         ones(time_steps)
     )
-    return Deterministic(component, label, resolution, initial_time, data)
+    return Deterministic(component, label, Dates.Minute(resolution), initial_time, data)
+end
+
+"""Constructs Deterministic after constructing a TimeArray from initial_time and time_steps.
+"""
+function Probabilistic(component::Component,
+                       label::String,
+                       resolution::Dates.Period,
+                       initial_time::Dates.DateTime,
+                       quantiles::Vector{Float64},
+                       time_steps::Int)
+
+    data = TimeSeries.TimeArray(
+        initial_time : Dates.Hour(1) : initial_time + resolution * (time_steps-1),
+        ones(time_steps, length(quantiles))
+    )
+
+    return Probabilistic(component, label, Dates.Minute(resolution), initial_time, quantiles, data)
+end
+
+"""Constructs Probabilistic Forecast after constructing a TimeArray from initial_time and time_steps.
+"""
+function Probabilistic(component::Component,
+                       label::String,
+                       quantiles::Vector{Float64},  # Quantiles for the probabilistic forecast
+                       data::TimeSeries.TimeArray)
+
+    if !(length(TimeSeries.colnames(data)) == length(quantiles))
+        error("The size of the provided quantiles and data columns is incosistent")
```

Comment:
typo:  inconsistent

Also, we should have a discussion about consistency when using exceptions.  I've been using our own exception or Julia exceptions.  One of the Base exceptions might be more appropriate here.  There is a debate here:  https://discourse.julialang.org/t/please-stop-using-error-and-errorexception-in-packages-and-base/12096

Reply:
Ok, I can use your exception. I wasn't aware of this debate.

Reply:
see 2a458fa